src/SymbolTable.jl

```diff
@@ -112,24 +126,26 @@ current_module()::Module = first(SYMBOL_TABLE)
 # scope.
 
 function enter_scope!()::Nothing
-    push!(current_scopes(), Scope())
+    push!(scopes_within_module(), Scope())
     return nothing
 end
 
 function exit_scope!()::Nothing
-    pop!(current_scopes())
-    @assert !isempty(current_scopes()) "Exited global scope. This shouldn't happen before leaving the module!"
+    pop!(scopes_within_module())
+    @assert !isempty(scopes_within_module()) "Exited global scope. This shouldn't happen before leaving the module!"
     return nothing
 end
 
-global_scope()::Scope = last(current_scopes())
-current_scope()::Scope = first(current_scopes())
+global_scope()::Scope = last(scopes_within_module())
+current_scope()::Scope = first(scopes_within_module())
 
 """
 Check if an item (the identifier in the node) is declared in any scope in the
 current module.
 """
-is_declared(node::Item)::Bool = any(scp -> node ∈ scp, current_scopes())
+is_declared(node::Item)::Bool = !isempty(SYMBOL_TABLE) ? any(scp -> node ∈ scp, scopes_within_module()) : false
```

Comment:
Can this be rewritten to:
```suggestion
is_declared(node::Item)::Bool = !isempty(SYMBOL_TABLE) && any(scp -> node ∈ scp, scopes_within_module())
```